wolfcrypt/src/sha256.c

```diff
@@ -1567,17 +1567,21 @@ static int InitSha256(wc_Sha256* sha256)
     {
         int ret = 0;
 
-        if (sha224 == NULL)
+        if (sha224 == NULL) {
             return BAD_FUNC_ARG;
+        }
+
+        XMEMSET(sha224, 0, sizeof(wc_Sha224));
```

Comment:
Missed wc_InitSha256_ex()

Reply:
@SparkiDev  thanks for your review! Would you recommend that I do a `XMEMSET` in each of these?

![image](https://user-images.githubusercontent.com/13059545/218228929-2bbf2f73-2c67-425a-acc0-80d15031b456.png)

I'm not entirely convinced it is a good idea for software-only hashes. The only reason I found this and started to do the initializations was due to undesired hardware encryption state data being _copied_ into a new sha `ctx` instance. 

Please advise. Thank you.

Reply:
If the PR is to memset the data structure when using hardware and they are the implementations for hardware, then yes.
If you can't find a common for each function, then it needs to be done in each.